yaksh/views.py

```diff
@@ -1796,6 +1796,69 @@ def download_assignment_file(request, quiz_id, question_id=None, user_id=None):
 
 @login_required
 @email_verified
+def upload_users(request, course_id):
+    user = request.user
+    ci = RequestContext(request)
+    course = get_object_or_404(Course, pk=course_id)
+    context = {'course': course}
+
+    if not (course.is_teacher(user) or course.is_creator(user)):
+        msg = 'You do not have permissions to this course.'
+        return complete(request, reason=msg)
+    if request.method == 'POST':
+        if 'csv_file' not in request.FILES:
+            context['message'] = "Please upload a CSV file."
+            return my_render_to_response('yaksh/course_detail.html', context,
+                    context_instance=ci)
+        csv_file = request.FILES['csv_file']
+        if not is_csv(csv_file):
+            context['message'] = "The file uploaded is not a CSV file."
+            return my_render_to_response('yaksh/course_detail.html', context,
+                    context_instance=ci)
+        headers = {'firstname':'', 'lastname':'', 'email':''}
+        reader = csv.DictReader(csv_file.read().decode('utf-8').splitlines())
+        if not headers_present(reader, headers):
+            context['message'] = "The CSV file does not contain the required headers"
+            return my_render_to_response('yaksh/course_detail.html', context,
+                    context_instance=ci)
+        context['upload_details'] = _read_user_csv(reader, headers, course)
+    return my_render_to_response('yaksh/course_detail.html', context,
+            context_instance=ci)
+
+
+def _read_user_csv(reader, headers, course):
+    upload_details = ["Upload Summary:"]
+    counter = 0;
+    for row in reader:
+        counter += 1
+        email, first_name, last_name = map(str.strip, [row[headers['email']],
+            row[headers['firstname']], row[headers['lastname']]])
+        if not email or not first_name or not last_name:
+            upload_details.append("{0} -- Missing Values".format(counter))
+            continue
+        user = User.objects.filter(email=email)
+        if user.exists():
+            upload_details.append("{0} -- {1} -- Email Already Exists".format(
+                counter, email))
+        else:
+            try:
+                user = User.objects.create_user(username=email, password=email,
+                        email=email, first_name=first_name, last_name=last_name)
+            except IntegrityError:
```

Comment:
Just to be on safe side you can do django import as `from django.db import IntegrityError`

Reply:
We literally impose no checks on the email id format,. We can have a regex validator, to check if the email is legit or not

Reply:
@adityacp , I agree, this will optimize the code for large data, but still need to check completely.
Changes done.

@maheshgudi 
Not validated email id because even if we validate with regex, there can be email ids with correct structure but in reality they do not exists . We are providing update email option and also I am setting email_verified True as these are been uploaded by the authenticated user that is moderator, I feel its fine if we do not validate for now.